core/src/main/java/org/apache/calcite/rel/metadata/BuiltInMetadata.java

```diff
@@ -83,9 +83,9 @@ public interface UniqueKeys extends Metadata {
      * represented as an {@link org.apache.calcite.util.ImmutableBitSet}, where
      * each bit position represents a 0-based output column ordinal.
      *
-     * <p>Note that a key plus other columns is still unique. Therefore a table
-     * with all columns unique has a unique key consisting of the empty set.
-     * This is the case with all single-row tables.
+     * <p>Note that a unique key plus other columns is still unique.
+     * Conversely, all columns are unique in a table with a unique key
+     * consisting of the empty set, as is the case for all single-row tables.
```

Comment:
Re-reading, I guess your comment is true, but it is extremely misleading. Only 0 and 1 row tables have `()` as a key. From that it follows that all rows are unique, all columns are unique, and all combinations of columns are unique. But we knew that, because the table had 0 or 1 rows.

Reply:
I changed the wording to accommodate the the zero-column case that you mentioned - all zero-or-one-row tables have `()` as a key, but not all tables that have all columns unique do.

However, I think it is true that all (even multi-row) tables with one or more columns that have all columns unique have `()` as a unique key. So I might be missing another point.

Reply:
x is a key if, for two rows, their values of x will be different if are not the same row.

(x, y) is a key if, for two rows, their values of x and y will be the different if they are not the same row.

() is a key if, for two rows the value of () will be different if they are not the same row. But () is a constant. So the table cannot have more than one row.

Sorry to keep hammering on this. But it's like trying to discuss logic if we disagree on the meaning of true and false. (The above is closely related to the finding in logic that "false implies anything".)

Reply:
No, this is good. I see your point now. In retrospect, your zero-column argument should have been sufficient. My confusion went like this:
1. If (A) is a unique key, (A,B) is also a unique key
2. A 0-or-1-row table has () as a unique key
3. All columns in a table with () as a key are unique
4. A table with all columns unique has () as a key

I didn't catch the leap between 3 and 4. I've been quietly unsettled about empty keys and this clarification helps with that.